docs/useful-snippets/snippets.rst

```diff
@@ -58,3 +58,57 @@ code as follows.
 If you would still like to run your code in Jupyter you will need to adjust the cell as noted above. Then you can
 you can add the `nbmultitask library <(https://nbviewer.org/github/micahscopes/nbmultitask/blob/39b6f31b047e8a51a0fcb5c93ae4572684f877ce/examples.ipynb)>`__
 or look at this `stackoverflow <https://stackoverflow.com/questions/50937362/multiprocessing-on-python-3-jupyter>`__.
+
+Using multiple schedules and data collectors
+-------------
+
+One may occasionally want multiple schedules with distinct sets of agents and collect data on each of those schedules independently while allowing the agents to interact in the same model. This would be particularly useful if the agents in your model do not all have the same attributes and you want to collect data on them in groups. You can create a simple scheduler class to mix the agents in each schedule and data collect on each schedule independently.
+
+.. code:: python
+
+  from random import shuffle
+
+  # simple schedule mixer
+  class ScheduleMixerRandom:
+      """
+      Be careful: ScheduleMixerRandom as written here would not capture changes to
+      the agent lists in the sub-schedules. The user would need to be mindful if
+      this functionality were to be included.
+      """
+
+      def __init__(self, schedules):
+          self._agents = {}
+          for s in schedules:
+          for a in s.agents:
+              self._agents[a.unique_id] = a # each agent must have a unique id across models
```

Comment:
Needs 2 spaces before the `#`. The code snippet should be formatted with Black before put in the snippets.rst.

Reply:
I ran Black but it said it didn't format RST files. Should I run just the Python through the formatter after pasting it into another file?

Reply:
> Should I run just the Python through the formatter after pasting it into another file?

Yes that works!

You can also try [blacken-docs](https://github.com/adamchainz/blacken-docs), that should format code blocks in Markdown and ReStructured Text.